added a new table option skipSchemaSync to prevent schema-builder to …

apezel opened the pull request:
Added a new skipSchemaSync option to @Table.
Useful to prevent schema builder to sync a table, for example if it's a view.

Example of usage : 

```javascript
@Table("view", {skipSchemaSync: true})
export class View {

    @PrimaryColumn()
    id: number;

    @Column()
    title: string;

}
```

*Added a test case in functional/connection/connection.ts with a new entity named View* #86

pleerock commented on src/decorator/tables/ClassTableChild.ts:
```diff
@@ -11,7 +11,8 @@ export function ClassTableChild(tableName?: string, options?: TableOptions) {
             type: "class-table-child",
-            orderBy: options && options.orderBy ? options.orderBy : undefined
+            orderBy: options && options.orderBy ? options.orderBy : undefined,
+            skipSchemaSync: (options && options.skipSchemaSync === true) || false
```
`options && options.skipSchemaSync === true` already returns boolean, so `|| false` is redundant here and only brings more confusion. But its not important.

apezel replied:
Ok, i can change it but ```options && options.skipSchemaSync === true```returns a boolean only if options isn't undefined. If options is undefined it returns undefined, not false

pleerock replied:
oh then maybe simply do `!!(options && options.skipSchemaSync === true)` ?

pleerock commented on src/schema-builder/SchemaBuilder.ts:
```diff
@@ -95,19 +95,23 @@ export class SchemaBuilder {
     // -------------------------------------------------------------------------
 
+    protected get entityToSyncMetadatas(): EntityMetadataCollection {
+        return <EntityMetadataCollection>this.entityMetadatas.filter( metadata => !metadata.table.skipSchemaSync );
```
can you reformat it this way: `<EntityMetadataCollection> this.entityMetadatas.filter(metadata => !metadata.table.skipSchemaSync)`

pleerock replied:
also your code is not correct. `.filter` does not return instance `EntityMetadataCollection`, it returns you `Array`.  So you are lying to compiler by casting it to `EntityMetadataCollection` will bring you errors once there will be a call to a specific method.

apezel replied:
Btw, the linter doesn't complain about spaces. But i'll change it too.

pleerock commented on test/functional/connection/connection.ts:
```diff
@@ -351,5 +353,19 @@ describe("Connection", () => {
+            await connection.syncSchema(true);
+            const queryRunner = await connection.driver.createQueryRunner();
+            let schema = await queryRunner.loadSchemaTables(["view"], new DefaultNamingStrategy());
+            expect(!schema.some( table => table.name === "view" )).to.be.true;
```
I think `!schema.some( table => table.name === "view" )` expression can be written in more human readable way (because I cant easily understand what this code does)

pleerock commented:
thank you for your PR. can you please review comments once?

apezel commented:
I added a filter function directly on EntityMetadataCollection, removed the spaces arround block functions and changed ```options && options.skipSchemaSync```

pleerock commented:
thanks ;)
